packages/bpk-component-datepicker/src/BpkDatepicker.js

```diff
@@ -70,6 +73,9 @@ class BpkDatepicker extends Component {
     if (this.props.onDateSelect) {
       this.props.onDateSelect(dateObj);
     }
+    if (this.props.onClose) {
+      this.props.onClose();
+    }
```

Comment:
Why do we need to call this in handleDateSelect, wondering when date selected, open status should rely on state not props change? Looking at stories, it seems like passing a new props to close date picker, but date picker is also  setting isOpen to false at L70-72.

Reply:
The onClose prop is not really for closing the date picker, it's a callback function that gets called when the date selector pop-up gets closed. This allows consumers to fully control the open/close state from the outside. I also added an use example there: https://github.com/Skyscanner/backpack/pull/1981/files#diff-9cb602e1e5145c854adc323bec617145R212

Does that answer the question?

Reply:
I see what you mean, the state and prop can get de-synchronised with only the onClose hook. I changed it to a onOpenChange hook instead so we can keep them in sync